include/broker/detail/master_actor.hh

```diff
@@ -70,21 +68,17 @@ public:
 
   void operator()(clear_command&);
 
-  caf::event_based_actor* self;
-
-  std::string id;
-
   topic clones_topic;
 
   backend_pointer backend;
 
-  caf::actor core;
-
   std::unordered_map<caf::actor_addr, caf::actor> clones;
 
-  endpoint::clock* clock;
+  std::unordered_map<data, publisher_id> publishers;
```

Comment:
That's keeping track of who published any given key in the current store?  I wonder if storing that state is even needed.  In the two cases that info gets used, expiration and snapshots, it may make more sense to identify the master store itself as the publisher, not the original/latest publisher that inserted/modified the key?

Looked like that info is used in two case: expiration and snapshots.  For either of those,

Reply:
The nice thing about keeping this map is that we can better keep track of who-changed-what last in the subscribers. However, I don't have strong feelings one way or another. Is your concern that his map could become too large?

Reply:
Yeah, partly thinking the map could become large -- it's replicating the entire key-space.  Other thought is that currently doesn't generalize for use with persistent storage backends.  Could likely do a larger change to the various backends to additionally embed the `publisher_id` along with associated value for a given key to optimize/solve both concerns, but wonder if we really intend to make good use of that info in first place.

Reply:
Ok, I've removed the `publishers` map from the master.